File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    vector<string> list = {"aya","ye","woo","ma"};
    for(auto& elem : babbling)
    {
        for(auto target : list)
        {
            size_t pos = elem.find(target);
            if(pos != string::npos)
                elem.replace(pos,target.length(),"@");
        }
        size_t pos = elem.find("@");
        while(pos != string::npos)
        {
            elem.replace(pos,1,"");
            pos = elem.find("@", pos);
        }
        
        if(elem.length() == 0)
            answer++;
    }
    return answer;
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cs
using System;

public class Solution {
    public int solution(string[] babbling) {
        int answer = 0;
        String[] list = new String[]{"aya","ye","woo","ma"};
        foreach(String s in babbling)
        {
            String temp = s;
            foreach(String target in list)
                temp = temp.Replace(target, " ");
            
            temp = temp.Trim();
            if(temp.Length == 0)
                answer++;
        }
        return answer;
    }
}